src/Microsoft.AspNet.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -533,26 +533,50 @@ private void RenderEndTagHelpersScope()
                                                   _tagHelperContext.ScopeManagerEndMethodName);
         }
 
-        private void RenderWriteTagHelperMethodCall(TagHelperChunk chunk)
+        private void RenderTagHelperOutput(TagHelperChunk chunk)
         {
+            var tagHelperOutputAccessor =
+                $"{ExecutionContextVariableName}.{_tagHelperContext.ExecutionContextOutputPropertyName}";
+
+            if (ContainsChildContent(chunk.Children))
```

Comment:
> We can only do this at compile time because we know if there's whitespace etc. that exist within the child content.

I was saying "Why bother with this added complexity at all?" Still need to write output even if it's only whitespace.

> Several self-closing examples

I said self-closing was covered. It's Razor comments, complex statement blocks, templates, ... inside tag helpers that have poor coverage. No way to know if we're missing a case where `Children` contains something obscure that will no longer render. Again, why bother with these special cases?